common/http/src/main/java/io/helidon/common/http/Http1HeadersParser.java

```diff
@@ -67,7 +80,22 @@ public static WritableHeaders<?> readHeaders(DataReader reader, int maxHeadersSi
             reader.skip(2);
             maxLength -= eol + 1;
 
-            headers.add(Http.Header.create(header, value));
+            if (parserMode == ParserMode.RELAXED && header == HeaderEnum.CONTENT_TYPE) {
```

Comment:
This is not the right place to resolve this problem. The parsing should be as fast as possible, even if the value is wrong.
The value should be resolved and validated when needed - e.g. in the case of `ContentType`, when somebody calls the content type method on server request, or client response, where we map the value to actual `MediaType` (or `HttpMediaType`)

Reply:
I'm reverting whole class (+ all required changes) to main version.